src/Chassis/Env/EnvSwitcher.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Collections.Generic;
+using Autofac;
+
+namespace Chassis.Env
+{
+    public class EnvSwitcher<TService>
+        where TService : class
+    {
+        readonly List<Type> _types = new List<Type>();
+        readonly Dictionary<AppEnv, Type> _maps = new Dictionary<AppEnv, Type>();
+
+        public void On<TType>(AppEnv env)
+            where TType : class
+        {
+            _types.Add(typeof(TType));
+            _maps.Add(env, typeof(TType));

```

Comment:
Do you want to check before adding if `env` already exists in `_maps` or just let the dictionary exception throw?